### PR TITLE
[1/2] delete useBlockTraceOnQueryResult

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

`useBlockTraceOnQueryResult` is now the default behavior when using our wrapped `useQuery`. To opt out you should set `blocking: false`

used this script to find useQuery calls without a corresponding useBlockTraceOnQueryResult call https://gist.github.com/salazarm/4d263ff119737695b944852df97f2723
Which surfaced these sites:

```
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/AssetDefinedInMultipleReposNotice.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetChoosePartitionsDialog.tsx: warningsResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/OverdueTag.tsx: result
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/useReportEventsModal.tsx: assetPartitionDefResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/instance/useDaemonStatus.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/jobs/JobsPageContent.tsx: queryResultOverview
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigEditorConfigPicker.tsx: queryResultNoAssets
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigEditorConfigPicker.tsx: queryResultAssets
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/partitions/BackfillMessaging.tsx: queryData
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/partitions/OpJobPartitionsView.tsx: currentQueryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/runs/useCursorPaginatedQuery.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/typeexplorer/TypeExplorerContainer.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceJobsRoot.tsx: queryResultOverview
```

I manually examined these and determined that either 
(1) They mistakenly were not blocking. 
(2) They were not blocking because they were part of a dialog so they don't affect page load anyways and blocking at them won't hurt (Might help in the long run if a component gets moved to be used during page load.
or (3) They actually should not be blocking since they're not a critical UX (eg. warnings)

## How I Tested These Changes

buildkite.

## Changelog [New | Bug | Docs]
NOCHANGELOG

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/app/Permissions.tsx [hunk 1]
```diff
@@ -135,11 +134,10 @@ export const PermissionsContext = React.createContext<PermissionsContextType>({
 
 export const PermissionsProvider = (props: {children: React.ReactNode}) => {
   const queryResult = useQuery<PermissionsQuery, PermissionsQueryVariables>(PERMISSIONS_QUERY, {
-    fetchPolicy: 'cache-first', // Not expected to change after initial load.
+    fetchPolicy: 'cache-first', // Not expected to change after initial load.'
```
Comment (alangenfeld) [inline js_modules/dagster-ui/packages/ui-core/src/app/Permissions.tsx:137]: typo ?

**Review by alangenfeld — COMMENTED** [review]
what gives you confidence that you got `{blocking: false},` in all the right places?

**Comment by salazarm** [comment]
@alangenfeld I wrote a script https://gist.github.com/salazarm/4d263ff119737695b944852df97f2723
Which surfaced these sites:

```

/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/AssetDefinedInMultipleReposNotice.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetChoosePartitionsDialog.tsx: warningsResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/OverdueTag.tsx: result
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/assets/useReportEventsModal.tsx: assetPartitionDefResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/instance/useDaemonStatus.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/jobs/JobsPageContent.tsx: queryResultOverview
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigEditorConfigPicker.tsx: queryResultNoAssets
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigEditorConfigPicker.tsx: queryResultAssets
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/partitions/BackfillMessaging.tsx: queryData
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/partitions/OpJobPartitionsView.tsx: currentQueryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/runs/useCursorPaginatedQuery.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/typeexplorer/TypeExplorerContainer.tsx: queryResult
/Users/marcosalazar/code/dagster/js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceJobsRoot.tsx: queryResultOverview
```

I manually examined these and determined that either 
(1) They mistakenly were not blocking. 
(2) They were not blocking because they were part of a dialog so they don't affect page load anyways and blocking at them won't hurt (Might help in the long run if a component gets moved to be used during page load.
or (3) They actually should not be blocking since they're not a critical UX (eg. warnings)

**Review by hellendag — APPROVED** [review]

**Comment by alangenfeld** [comment]
nice 👍

**Pull request merged**